IdentityCore/src/cache/MSIDTokenCacheKey.m

```diff
@@ -106,11 +87,17 @@ + (MSIDTokenCacheKey *)keyWithAuthority:(NSURL *)authority
                                resource:(NSString *)resource
                                     upn:(NSString *)upn
 {
-    return [[MSIDTokenCacheKey alloc] initWithAccount:upn
-                                              service:[self.class serviceWithAuthority:authority
-                                                                              resource:resource
-                                                                              clientId:clientId]
-                                                 type:nil];
+    MSIDTokenCacheKey *key = [[MSIDTokenCacheKey alloc] initWithAccount:upn
+                                                                service:[self.class serviceWithAuthority:authority
+                                                                                                resource:resource
+                                                                                                clientId:clientId]
+                                                                   type:nil];
+    
+    key.authority = authority;
```

Comment:
Two things here:
1. Can we make those properties readonly for other classes?
2. I'm a bit worried about setting those properties only in one keyWith... method, because someone will decide to use another key method and those properties will be nil...

Reply:
1. authority, resource, clientId are already private properties. 
2. Agree, lets discuss option of parsing it from/to 'service' so we will no need to store authority, resource, clientId at all.

Reply:
Please add a Github issue for nr 2.

Reply:
added #54